### DISCUSSION TITLE
How to deloy examples/i18n-routing/ to firebase hosting?

### DISCUSSION THREAD

**Discussion by paulghc051080** [vercel/next.js#39418] [Help]
I have clone and test the nextjs i18n-routing example in local machine which operating system is win10. It is working fine with no issues. After that I tried to deploy the app to Firebase hosting, the result is shown index page with 404 error, but I can manually to change the url with locales. It can show page but not all the features works properly . This is my Firebase app url https://asnext-f2e94.web.app/. 

The Firebase deploy method I was following the instructions from nextjs Firebase hosting example code. I also tried that example and deployed successfully. However, is there has example code for nextjs i18n-routing deploy to Firebase hosting example can follow?

**Comment by MiftahClassifieds** [comment]
same issue firebase hosting could not route pages site/en or site/fr . always giving 404 err

**Comment by Nadeera3784** [comment]
Hi, @paulghc051080,
Did you update the firebase.json config? 
```

{
  "hosting": {
    "public": "out",
    "rewrites": [
      {
        "source": "**",
        "destination": "/index.html"
      }
    ]
  }
}

```
And also the next.config.js

```
// next.config.js
module.exports = {
  basePath: '/your-public-url-path'
  // rest of your config
}

```
